packages/realm-server/tests/realm-server-test.ts

```diff
@@ -3012,6 +3012,27 @@ module('Realm Server', function (hooks) {
         ],
       });
     });
+
+    test('can fetch public realms', async function (assert) {
```

Comment:
Would be good to add a test for when one of the realm info fetches fails during executing this request

Reply:
Not sure how to do that in realm server test. Is there a way to mock an API like `_info` so we can make the response failed?

Reply:
I think you can add a handler in virtual network that is passed to the server and to the realm, and mount a handler that returns an error if it encounters a specific url. Perhaps something like this:

<img width="441" alt="image" src="https://github.com/user-attachments/assets/2b00b454-debf-4f93-9ed0-192e60ae962d">